### PR TITLE
fix(curriculum): delineate span elements in typography

### LINKED ISSUE #51786: Typographyproject: Step-43 Misleading  instructions
### Describe the Issue

To begin with, the directions implied that there would be 3 spans that weren't interconnected for alignment.

My initial assumption would be that the solution would be something like:

```html 
 <p><span class="bold">Total Fat </span><span>8g</span><span class="bold">10%</span> 
```

Nowhere did it seem to imply that a span was supposed to be nested inside another span. But when I looked at the code for the next challenge, that became clear. However, I ran into another problem; spacing of my elements will cause the tests to fail. I like my opening and closing tags to be on separate lines usually, but here it will break the test. The tests need a little work. 

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-typography-by-building-a-nutrition-label/step-43

### Your code

```
   <div class="daily-value small-text">
      <p class="bold right">% Daily Value *</p>
      <div class="divider"></div>
     <p>
       <span>
         <span class="bold"> 
           Total Fat  
           </span> 
           8g</span>
         <span class="bold">
           10%
          </span>
      </p>
    </div>
```

### Expected behavior

I expect the instructions to clarify that the span for "Total Fat" is supposed to be nested inside another span where "8g" is supposed to be. Additionally I'd like the tests to be a little more spacing resistant so that if the right answer is rendered, the tests don't fail. 

### Screenshots

My solution

### System

- Device: Desktop Computer
- OS: Windows 10 
- Browser: Firefox
- Version: 118.0.1 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Supravisor**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #51786

image

image

image

image

image

### CODE REVIEW

**Comment by anishlukk123** [comment]
please resolve the failing tests

**Comment by ilenia-magoni** [comment]
@anishlukk123 the failing tests need to be resolved after review. Once the PR is approved, then the new steps can be propagated to other languages

File: curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/667d1bb875f5961913176069.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+Notice how the text `8g` appears centered in the preview.
+
+Nest the `Total Fat` `span` element and the text `8g` in an additional `span` element for alignment.
+
+# --hints--
+
+Nest the `span` element with the class of `bold` and text `Total Fat` and the text `8g`.
+
+```js
+assert(document.querySelector('.daily-value.small-text')?.lastElementChild?.querySelector('span > span')?.innerText === 'Total Fat');
```
Comment (moT01) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/667d1bb875f5961913176069.md:19]:
```suggestion
assert.match(document.querySelector('.daily-value.small-text')?.lastElementChild?.querySelector('span')?.innerHTML, /^<span class=("|')bold\1>Total Fat<\/span> 8g\s?$/);
```

This test wasn't work quite right. It was passing by only checking Total Fat `span`. I came up with this.

File: curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f7de4487b64919bb4aa5e.md [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
-dashedName: step-53
+dashedName: step-54
 ---
 
 # --description--
 
-After your last `.divider`, create a new `p` element with the text `Cholesterol 0mg 0%`. Wrap the text `Cholesterol` in a `span` element, and give that `span` element the `class` of `bold`. Wrap the text `0%` in another `span` element, with the `class` of `bold`. Finally, nest the `Cholesterol` `span` element and the text `0mg` in an additional `span` element for alignment.
+After your last `.divider`, create a new `p` element with the text `Cholesterol 0mg 0%`. Wrap the text `Cholesterol` in a `span` element, and give that `span` element the `class` of `bold`. Wrap the text `0%` in another `span` element, with the `class` of `bold`.
+
+Nest the `Cholesterol` `span` element and the text `0mg` in an additional `span` element for alignment.  This final part will nest a `span` element within a `span` element.
```
Comment (moT01) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f7de4487b64919bb4aa5e.md:12]:
```suggestion
After your last `.divider`, create a new `p` element with the text `Cholesterol 0mg 0%`. Then, wrap the text `Cholesterol` in a `span` element, `0%` in another, and give each of them a class of `bold`.

Finally, nest the `span` element containing the text `Cholesterol` along with the text `0mg`, in an additional `span` element for alignment.
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f7e7281626a92bbd62da8.md [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+dashedName: step-55
 ---
 
 # --description--
 
-Below your last `p` element, create another `p` element with the text `Sodium 160mg 7%`. Wrap the text `Sodium` in a `span` element with a `class` attribute set to `bold`. Wrap the `7%` text in another `span` element with the `class` set to `bold`. Also add an additional `span` element around `Sodium 160mg` for aligning it correctly.
+Below your last `p` element, create another `p` element with the text `Sodium 160mg 7%`. Wrap the text `Sodium` in a `span` element with a `class` attribute set to `bold`. Wrap the `7%` text in another `span` element with the `class` set to `bold`.
+
+Also add an additional `span` element around `Sodium 160mg` for alignment.
+As you did in the previous step, this will nest a `span` element within a `span` element.
```
Comment (moT01) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f7e7281626a92bbd62da8.md:13]:
```suggestion
Below your last `p` element, create another `p` element with the text `Sodium 160mg 7%`. Put `sodium` and `7%` each in their own `span` with a class of a `bold` like you did with the others.

Then, add an additional `span` element around `Sodium 160mg` for alignment again.
```

File: curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f7ecb09de9a938ef94756.md [hunk 4]
```diff
@@ -1,13 +1,15 @@
...
-dashedName: step-55
+dashedName: step-56
 ---
 
 # --description--
 
-Below your last `p` element, add another `p` element with the text `Total Carbohydrate 37g 13%`. Like before, use `span` elements to make the text `Total Carbohydrate` and `13%` bold. Also add an additional `span` element around the `Total Carbohydrate 37g` text to have it aligned to the left and `13%` to the right.
+Below your last `p` element, add another `p` element with the text `Total Carbohydrate 37g 13%`. Like before, use `span` elements to make the text `Total Carbohydrate` and `13%` bold. 
+
+Also add an additional `span` element around the `Total Carbohydrate 37g` text to have it aligned to the left and `13%` to the right.  Here you will nest a `span` element within a `span` element.
```
Comment (moT01) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-typography-by-building-a-nutrition-label/615f7ecb09de9a938ef94756.md:12]:
```suggestion
Below your last `p` element, add another `p` element with the text `Total Carbohydrate 37g 13%`. Like before, use `span` elements to make the text `Total Carbohydrate` and `13%` bold. Then, wrap the nutrient and amount in a `span` for alignment again.
```

**Review by moT01 — CHANGES_REQUESTED** [review]
This is looking good @Supravisor 👍 

I left a number of grammar suggestions to try and make things a little more concise, while hopefully providing enough for it to be clear to campers what to do - mainly, how I would have liked to see it written - so it's kind of just my preference. Feel free to bring up any questions or comments on any of them.

**Review by moT01 — APPROVED** [review]
LGTM @Supravisor 🎉  Hopefully this is clear enough.

Note, we will need to run a curriculum sync after this is merged.

**Review by Sembauke — APPROVED** [review]

**Comment by Sergeantkalpesh** [comment]
freecodepng

**Pull request merged**